### PR TITLE
Add a traject setting that allows the component identifiers to be configured

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

**Comment by jcoyne** [comment]
This seems like a good change. 👍

**Comment by seanaery** [comment]
I like it! Thanks for the quick work on this issue.

File: lib/arclight/traject/ead2_config.rb [hunk 1]
```diff
@@ -56,6 +56,7 @@
   provide 'repository', ENV.fetch('REPOSITORY_ID', nil)
   provide 'logger', Logger.new($stderr)
   provide 'component_traject_config', File.join(__dir__, 'ead2_component_config.rb')
+  # provide 'component_identifier_format', '%{root_id}%{ref_id}'
```
Comment (jcoyne) [inline lib/arclight/traject/ead2_config.rb:59]: Cruft?

File: lib/arclight/traject/ead2_component_config.rb [hunk 2]
```diff
@@ -90,10 +93,15 @@
 end
 
 to_field 'id' do |_record, accumulator, context|
-  accumulator << [
-    settings[:root].output_hash['id'],
-    context.output_hash['ref_ssi']
-  ].join
+  next if context.output_hash['id']
```
Comment (jcoyne) [inline lib/arclight/traject/ead2_component_config.rb:96]: What is this for? Can it have some documentation?

File: lib/arclight/traject/ead2_component_config.rb [hunk 3]
```diff
@@ -68,6 +69,8 @@
 # NOTE: All fields should be stored in Solr
 # ==================
 to_field 'ref_ssi' do |record, accumulator, _context|
+  next if context.output_hash['ref_ssi']
```
Comment (jcoyne) [inline lib/arclight/traject/ead2_component_config.rb:72]: What is this for? Can it have some documentation?

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Comment by marlo-longley** [comment]
The work in this PR was captured in #1478.

**Pull request closed without merging**